crypto/tbs/src/poly.rs

```diff
@@ -1,117 +1,28 @@
-use std::fmt::Debug;
-use std::marker::PhantomData;
-use std::ops::{Add, AddAssign, Mul, MulAssign};
-use std::slice::Iter;
-
-use ff::Field;
+use bls12_381::{G2Affine, G2Projective, Scalar};
+use group::Curve;
 use rand::RngCore;
 
 use crate::FromRandom;
 
-#[derive(Debug)]
-pub struct Poly<G, S>
-where
-    G: Debug,
-{
-    coefficients: Vec<G>,
-    _pd: PhantomData<S>,
-}
-
-impl<G, S> Poly<G, S>
-where
-    G: Debug + MulAssign<S> + AddAssign<G> + Copy,
-    S: Copy,
-{
-    pub fn from(coefficients: Vec<G>) -> Self {
-        Poly {
-            coefficients,
-            _pd: PhantomData,
-        }
-    }
-
-    pub fn evaluate(&self, x: impl Into<S>) -> G {
-        let mut result = *self
-            .coefficients
-            .last()
-            .expect("Polynomial has no coefficients");
-        let x: S = x.into();
-        for &c in self.coefficients.iter().rev().skip(1) {
-            result.mul_assign(x);
-            result.add_assign(c);
-        }
-        result
-    }
-
-    pub fn coefficients(&self) -> Iter<G> {
-        self.coefficients.iter()
-    }
+pub fn random_scalar_coefficients(degree: usize, rng: &mut impl RngCore) -> Vec<Scalar> {
```

Comment:
Technically we probably also want to require `CryptoRng` to be implemented by the rng